pyproject.toml

```diff
@@ -0,0 +1,54 @@
+[build-system]
+requires = ["setuptools >= 77.0.3"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "willa"
+version = "0.0.1"
+dependencies = [
+    "pymarc",
+    "python-dotenv",
+    "requests",
+]
+requires-python = ">= 3.13"
+authors = [
+    {name = "Anna Wilcox"},
+    {name = "David Zuckerman", email = "dzuckerm@berkeley.edu"},
+    {name = "Jason Raitz"},
+]
+readme = "README.rst"
+license = "MIT"
+license-files = ["LICENSE"]
+
+[project.optional-dependencies]
+test = [
+    "requests-mock",
+]
+lint = [
+    "mypy ~= 1.17.1",
+    "pylint ~= 3.3",
+]
+dev = [
+    "willa[test,lint]"
+]
+
+[project.urls]
+Repository = "https://github.com/BerkeleyLibrary/willa"
+Issues = "https://github.com/BerkeleyLibrary/willa/issues"
+Changelog = "https://github.com/BerkeleyLibrary/willa/blob/main/CHANGELOG.rst"
+
+[tool.mypy]
+python_version = "3.13"
+warn_unused_configs = true
+warn_redundant_casts = true
+warn_return_any = true
+strict_equality = true
+check_untyped_defs = true
+disallow_subclassing_any = true
+# We can't enable disallow_untyped_calls because of PyMARC.
+disallow_untyped_calls = false
+disallow_incomplete_defs = true
+disallow_untyped_defs = true
```

Comment:
if we're adding `mypy` here, do we want to run typechecks as part of CI?

Reply:
I didn't enable them yet when I wrote this because without #8 we had a single MyPy error:

```
willa/tind/fetch.py:57: error: Incompatible types in assignment (expression has type "str | None", variable has type "str")  [assignment]
```

We should be clear to add `mypy` to CI now, and I can add that in a further MR.